pkg/engine/utils.go

```diff
@@ -166,6 +166,9 @@ func MatchesResourceDescription(
 	gvk schema.GroupVersionKind,
 	subresource string,
 ) error {
+	if resourceRef.Object == nil {
+		return fmt.Errorf("resource is empty")
+	}
```

Comment:
What is the `resourceRef` for DELETE requests?

Reply:
`resourceRef` is an `unstructured.Unstructured` object, the caller will pass either new resource or old resource.
From the point of view of matching it just doesn't make sense to match an object if it is empty.